Cargo.toml

```diff
@@ -1,6 +1,8 @@
 [workspace]
 
 members = [
+    "components/data_provider",
+    "components/data_provider_json",
```

Comment:
I know the situation with `-` vs `_` in Rust is quirky, but I think we generally use `-` in directories and names, and then use `_` only when importing.

Examples:
* https://doc.rust-lang.org/book/ch07-01-packages-and-crates.html
* https://doc.rust-lang.org/cargo/guide/project-layout.html

Since we use `icu-locale`, `icu-data-provider` etc. I think the directory should be `components/data-provider` and `components/data-provider-json`.